skbio/metadata/_interval.py

```diff
@@ -0,0 +1,699 @@
+                _assert_valid_boundary(boundary)
+            self._boundaries = value
+
+    @property
+    @experimental(as_of='0.5.0-dev')
+    def locations(self):
+        '''The coordinates of the interval feature.
+
+        It should be a list of tuples of int pair. Each tuple stores
+        the start and end coordinates of a span of the interval feature.
+        '''
+        if self.dropped:
+            raise RuntimeError(
+                'Cannot retrieve locations from dropped Interval object.')
+        return self._locations
+
+    @locations.setter
+    @experimental(as_of='0.5.0-dev')
+    def locations(self, value):
+        '''Set ``locations``.
+
+        WARNING: setting ``locations`` will reset ``boundaries`` value to True.
+        This is not totally surprising because it is justifiable your old
+        boundaries don't fit the new locations.
+        '''
+        if self.dropped:
+            raise RuntimeError(
+                'Cannot change locations to dropped Interval object.')
+        # check iterability
+        try:
+            value = list(value)
+        except:
+            raise TypeError(
+                'Cannot give a non-iterable (%r) to `locations`.' % value)
+        # check it is not empty
+        if len(value) == 0:
+            raise ValueError(
+                'Cannot give empty `location`.')
+        # check each contiguous span is in right format
+        for location in value:
+            _assert_valid_location(location)
+        # don't forget to sort the spans
+        self._locations = sorted(value)
+        # reset all the boundaries to True!!
+        self.boundaries = None
+        self._interval_metadata._is_stale_tree = True
+
+    @property
+    @experimental(as_of='0.5.0-dev')
+    def metadata(self):
+        '''The metadata of the interval feature.
+
+        It stores the metadata (eg. gene name, function, ID, etc.) of
+        the interval feature as a ``dict``.
+        '''
+        if self.dropped:
+            raise RuntimeError(
+                'Cannot retrieve metadata from dropped Interval object.')
+        return self._metadata
+
+    @metadata.setter
+    @experimental(as_of='0.5.0-dev')
+    def metadata(self, value):
+        if self.dropped:
+            raise RuntimeError('Cannot change metadata to dropped '
+                               'Interval object.')
+        if not isinstance(value, dict):
+            raise TypeError("metadata must be a dict, not %r" % value)
+        self._metadata = value
+
+    @property
+    @experimental(as_of='0.5.0-dev')
+    def dropped(self):
+        '''Boolean value indicating if the ``Interval`` object is dropped.
+
+        Notes
+        -----
+        This property is not writable.
+
+        See Also
+        --------
+        skbio.metadata.Interval.drop
+        '''
+        return self._interval_metadata is None
+
+
+class IntervalMetadata():
+    """Stores the interval features of a sequence.
```

Comment:
Generalize this docstring so it isn't specific to a sequence (using a sequence as an example is great though)